src/main/java/com/github/fge/jsonpatch/RemoveOperation.java

```diff
@@ -56,17 +62,60 @@ public JsonNode apply(final JsonNode node)
             return MissingNode.getInstance();
         if (path.path(node).isMissingNode())
             throw new JsonPatchException(BUNDLE.getMessage(
-                "jsonPatch.noSuchPath"));
+                    "jsonPatch.noSuchPath"));
         final JsonNode ret = node.deepCopy();
         final JsonNode parentNode = path.parent().get(ret);
-        final String raw = Iterables.getLast(path).getToken().getRaw();
+        final String fieldName = Iterables.getLast(path).getToken().getRaw();
+
         if (parentNode.isObject())
-            ((ObjectNode) parentNode).remove(raw);
+            removeFromObject(parentNode, fieldName);
         else
-            ((ArrayNode) parentNode).remove(Integer.parseInt(raw));
+            ((ArrayNode) parentNode).remove(Integer.parseInt(fieldName));
         return ret;
     }
 
+    private void removeFromObject(JsonNode parentNode, String fieldName) {
+        final JsonNode targetObject = parentNode.get(fieldName);
+        if (targetObject.isArray()) {
+            removeFromArray((ObjectNode) parentNode, fieldName, (ArrayNode) targetObject);
+        } else {
+            ((ObjectNode) parentNode).remove(fieldName);
+        }
+    }
+
+    private void removeFromArray(ObjectNode parentNode, String fieldName, ArrayNode targetObject) {
+        if (value.isNull()) {
+            parentNode.remove(fieldName);
+        }
+        else if (value.isArray()) {
+            final ArrayNode arr = remove(targetObject, (ArrayNode) value);
+            parentNode.replace(fieldName, arr);
+        } else {
+            final ArrayNode arr = remove(targetObject, value);
+            parentNode.replace(fieldName, arr);
+        }
+    }
+
+    private ArrayNode remove(ArrayNode arrayObject, JsonNode toRemove) {
+        ArrayNode reduced = arrayObject.deepCopy();
+        int i = 0;
+        for (JsonNode e : arrayObject) {
+            if (e.equals(toRemove)) {
+                reduced.remove(i);
+            }
+            i++;
```

Comment:
Does `reduced.remove(i);` have an impact on the current index `i`? Do we have to put `i++` into an else block maybe? I am not sure about that. But if you remove `i` twice, would it remove two consecutive items or would the second time fail? 🤔

Reply:
Not sure what you mean. Can you provide an example that I can put in [remove.json](https://github.com/collaborationFactory/cplace-json-patch/pull/2/files#diff-3556e0d4c9a1569c8840fcf119b5b59076d08fcc360cc4074702a3beb2f27f03)?

Reply:
```
  {
            "op": { "op": "remove", "path": "/x", "value": ["y", "z"] },
            "node": { "x": [ "y", "z", "a" ], "foo": "bar" },
            "expected": { "x": [ "a" ], "foo": "bar" }
        },
```